File: citizen-service/src/main/java/com/civicpulse/citizen/service/interfaces/CitizenService.java
package com.civicpulse.citizen.service.interfaces;

import com.civicpulse.citizen.dto.request.CreateCitizenRequest;
import com.civicpulse.citizen.dto.request.UpdateCitizenRequest;
import com.civicpulse.citizen.dto.response.CitizenResponse;
import com.civicpulse.citizen.util.enums.CitizenStatus;

import java.util.List;

public interface CitizenService {

    CitizenResponse registerCitizen(CreateCitizenRequest request);

    CitizenResponse getCitizenById(Long id);

    List<CitizenResponse> getAllCitizens();

    CitizenResponse updateCitizen(Long id,
                                  UpdateCitizenRequest request);

    void deleteCitizen(Long id);

    List<CitizenResponse> getCitizensByWard(String wardNumber);

    List<CitizenResponse> getCitizensByStatus(CitizenStatus status);

}

File: civicpulse-frontend/src/services/citizenService.js
import api from "../api/axios";

const BASE_URL = "/api/citizens";

const citizenService = {

    getAllCitizens: () => api.get(BASE_URL),

    getCitizenById: (id) => api.get(`${BASE_URL}/${id}`),

    registerCitizen: (data) => api.post(BASE_URL, data),

    updateCitizen: (id, data) => api.put(`${BASE_URL}/${id}`, data),

    deleteCitizen: (id) => api.delete(`${BASE_URL}/${id}`),

    getCitizensByWard: (ward) =>
        api.get(`${BASE_URL}/ward/${ward}`),

    getCitizensByStatus: (status) =>
        api.get(`${BASE_URL}/status/${status}`)
};

export default citizenService;